azure-kusto-data/source/client.js

```diff
@@ -86,62 +86,57 @@ module.exports = class KustoClient {
             payload = stream;
             clientRequestPrefix = "KNC.executeStreamingIngest;";
             headers["Content-Encoding"] = "gzip";
+            headers["Content-Type"] = "multipart/form-data";
         }
-
         headers["x-ms-client-request-id"] = clientRequestId || clientRequestPrefix + `${uuidv4()}`;
 
-        return this.aadHelper.getAuthHeader((err, authHeader) => {
-            if (err) return callback(err);
-
-            headers["Authorization"] = authHeader;
+        headers["Authorization"] = await this.aadHelper._getAuthHeader();
 
-            return this._doRequest(endpoint, headers, payload, timeout, properties, callback);
-        });
+        return this._doRequest(endpoint, headers, payload, timeout, properties);
     }
 
-    _doRequest(endpoint, headers, payload, timeout, properties, callback) {
-        return request({
-            method: "POST",
-            url: endpoint,
-            headers,
-            body: payload,
+    async _doRequest(endpoint, headers, payload, timeout, properties) {
+        let axiosConfig = {
+            headers: headers,
             gzip: true,
-            timeout
-        }, this._getRequestCallback(properties, callback)
-        );
+            timeout: timeout
+        };
+
+        let axiosResponse;
+        try {
+            axiosResponse = await axios.post(endpoint, payload, axiosConfig);
+        }
+        catch (error) {
+            if(error.response){
+                throw error.response.data.error;
```

Comment:
Do we always have data and error defined in such case?

Reply:
yes (from what I have checked)